src/aap_eda/analytics/analytics_collectors.py

```diff
@@ -163,8 +176,77 @@ def decision_environments_table(
 def event_streams_table(
     since: datetime, full_path: str, until: datetime, **kwargs
 ):
-    query = _get_query(models.EventStream.objects, since, until)
-    return _copy_table("event_streams", query, full_path)
+    event_streams = (
+        models.EventStream.objects.filter(
+            Q(created_at__gt=since, created_at__lte=until)
+            | Q(modified_at__gt=since, modified_at__lte=until)
+        )
+        .order_by("id")
+        .distinct()
+    ).values(
+        "id",
+        "organization_id",
+        "name",
+        "event_stream_type",
+        "eda_credential_id",
+        "uuid",
+        "created_at",
+        "modified_at",
+        "events_received",
+        "last_event_received_at",
+    )
+
+    query = (
+        str(event_streams.query)
+        .replace(_datetime_format(since), f"'{since.isoformat()}'")
+        .replace(_datetime_format(until), f"'{until.isoformat()}'")
+    )
+
+    return _copy_table(
+        "event_streams", f"COPY ({query}) TO STDOUT WITH CSV HEADER", full_path
+    )
+
+
+@register(
+    "event_streams_by_activation_table",
+    "1.0",
+    format="csv",
+    description="Data on event_streams used by each activation",
+)
+def event_streams_by_activation_table(
+    since: datetime, full_path: str, until: datetime, **kwargs
+):
+    activations = models.Activation.objects.filter(
+        Q(created_at__gt=since, created_at__lte=until)
+        | Q(modified_at__gt=since, modified_at__lte=until)
+    ).distinct()
+
+    event_streams = models.EventStream.objects.none()
+    for activation in activations:
+        event_streams |= activation.event_streams.all()
+
+    if not bool(event_streams):
+        return
+
+    event_streams = event_streams.extra(
```

Comment:
Using `extra` method is not recommended. Ref: https://docs.djangoproject.com/en/4.2/ref/models/querysets/#extra

We should be able to do using annotate as you do in https://github.com/ansible/eda-server/pull/1084/